core/tests/test_apps.cardano.bech32.py

```diff
@@ -0,0 +1,31 @@
+from common import *
+
+from apps.cardano.shelley.bech32 import bech32_encode, bech32_decode
+
+
+# todo: GK - other tests? Encode, Decode separately
+@unittest.skipUnless(not utils.BITCOIN_ONLY, "altcoin")
+class TestCardanoBech32(unittest.TestCase):
+    def test_decode_and_encode(self):
+        valid_bechs = [
+            # human readable part, bech32
+            ("a", "a12uel5l"),
+            ("an83characterlonghumanreadablepartthatcontainsthenumber1andtheexcludedcharactersbio",
+                "an83characterlonghumanreadablepartthatcontainsthenumber1andtheexcludedcharactersbio1tt5tgs"),
+            ("abcdef", "abcdef1qpzry9x8gf2tvdw0s3jn54khce6mua7lmqqqxw"),
+            ("1", "11qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqc8247j"),
+            ("split", "split1checkupstagehandshakeupstreamerranterredcaperred2y9e3w"),
+        ]
+
+        for valid_bech_tuple in valid_bechs:
+            valid_hrp = valid_bech_tuple[0]
```

Comment:
What's hrp? I prefer if variable names are clear to casual readers (e.g. the Trezor people).

Reply:
`hrp` is used by the bech32.py already included in the project, that's why I used it.  It's the human readable part of bech. And yes, I'll improve the naming consistency.